pkg/trace/stats/statsraw.go

```diff
@@ -106,15 +106,18 @@ type RawBucket struct {
 
 	// this should really remain private as it's subject to refactoring
 	data map[Aggregation]*groupedStats
+
+	containerTagsByID map[string][]string // a map from container ID to container tags
 }
 
 // NewRawBucket opens a new calculation bucket for time ts and initializes it properly
 func NewRawBucket(ts, d uint64) *RawBucket {
 	// The only non-initialized value is the Duration which should be set by whoever closes that bucket
 	return &RawBucket{
-		start:    ts,
-		duration: d,
-		data:     make(map[Aggregation]*groupedStats),
+		start:             ts,
+		duration:          d,
+		data:              make(map[Aggregation]*groupedStats),
+		containerTagsByID: getContainerTagsByIDFromPool(),
```

Comment:
I'm trying to think of a way that doesn't require callers to implicitly know about the pooled map. As it stands this could lead to subtle bugs if future callers aren't careful.

A few ideas come to mind:
1. Have the caller pass the map in explicitly, so that the lifespan of the map exists outside of the bucket.
2. Pool the buckets themselves.

Not super wild about either. At the end of the day, do we need the extra complexity relative to the performance gain? I think I'd be fine with the current organization + temp maps.

Reply:
Or 3, we move the bucket type and its API into the concentrator's code file and keep it all private so that everything can be managed in one place. Not sure if we use this type outside of this package anyway (took a quick look and it seemed like we don't).

Reply:
> Not super wild about either. At the end of the day, do we need the extra complexity relative to the performance gain? I think I'd be fine with the current organization + temp maps.

I feel the same way - given the negligible performance gain from the benchmark result + the maintenance overhead, it does not make too much sense to me to keep the cache / pool or do large refactor. I've reverted the pool commit, please let me know how you think.

Reply:
Sounds good, will have a look. Thanks for all the iteration here!